src/shared/credentials/userCredentialsUtils.ts

```diff
@@ -157,4 +153,19 @@ export class UserCredentialsUtils {
             return { isValid: false, invalidMessage: reason }
         }
     }
+
+    private static async createSTSClient(
```

Comment:
Nit: Would it make pass in the options directly?

```typescript
public static async validateCredentials(params: ServiceConfigurationOptions & { sts?: STS }) {
    ...
    sts = await UserCredentials.createSTSClient(params)
    ...
}

private static async createSTSClient(awsServiceOpts: ServiceConfigurationOptions) {
    return await ext.sdkClientBuilder.createAndConfigureServiceClient<STS>(
        (options) => new STS(options),
        awsServiceOpts,
    )
}
```

Also, at some point (not in this change), we should refactor static utility methods to standalone functions.